packages/beta/src/types.ts

```diff
@@ -304,3 +304,133 @@ export interface Subscription {
   subscriberExternalId?: CogniteExternalId;
   metadata?: Metadata;
 }
+
+export interface SimulatorUnitRecord {
+  label: string;
+  value: string;
+}
+export interface SimulatorUnitSystem {
+  label: string;
+  defaultUnits: Record<string, string>;
+}
+
+export interface SimulatorUnitsMap {
+  label: string;
+  units: SimulatorUnitRecord[];
+}
+
+export interface SimulatorUnits {
+  unitsMap?: Record<string, SimulatorUnitsMap>;
+  unitSystem?: Record<string, SimulatorUnitSystem>;
+}
+
+export interface SimulatorBoundaryCondition {
+  name: string;
+  address: string;
+  key: string;
+}
+
+export interface SimulatorModelType {
+  name: string;
+  key: string;
+}
+
+export interface SimulatorStepField {
+  name: string;
+  label: string;
+  info: string;
+}
+
+export interface SimulatorStep {
+  stepType: string;
+  fields: SimulatorStepField[];
+}
+export interface SimulatorResource {
+  id: CogniteInternalId;
+  externalId?: CogniteExternalId;
+  fileExtensionTypes: string[];
+  isBoundaryConditionsEnabled: boolean;
+  boundaryConditions: SimulatorBoundaryCondition[];
+  isCalculationsEnabled: boolean;
+  modelTypes: SimulatorModelType[];
+  enabled: boolean;
+  stepFields?: SimulatorStep[];
+  units?: SimulatorUnits[];
+  createdTime: Timestamp;
+  lastUpdatedTime: Timestamp;
+}
+
+export interface SimulatorResourceFilter {
+  externalIds?: CogniteExternalId[];
+}
+
+export interface SimulatorResourceFilterQuery extends FilterQuery {
+  filter?: SimulatorResourceFilter;
+}
+
+export interface SimulatorResourceCreate {
+  externalId?: CogniteExternalId;
+  fileExtensionTypes?: string[];
```

Comment:
```suggestion
  externalId: CogniteExternalId;
  fileExtensionTypes: string[];
```
top two aren't optional it seems